modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/BatchEngineExportTaskExecutorImpl.java

```diff
@@ -124,10 +124,6 @@ private void _exportItems(BatchEngineExportTask batchEngineExportTask)
 		UnsyncByteArrayOutputStream unsyncByteArrayOutputStream =
 			new UnsyncByteArrayOutputStream();
 
-		ZipOutputStream zipOutputStream = _getZipOutputStream(
```

Comment:
hi @shuyangzhou 
I didn't put zipOutputStream inside try block as every writer is closing outputStream in its close method so this move will not have an effect.

Reply:
@ivicac if the close happened successfully, yes, it makes no difference. But the try-with-resources is mainly ensuing everything is properly close under all circumstances. To ensure that, all IOs must be inside the try block statement list.

Reply:
ok, I was still thinking that because writer's close method will be always called as it is under try-with-resources block, it is enough to close zipOutputStream inside it, but understand what you want to tell me, and agree with that. This way nobody has to think where zipOutputStream is closed. But then it is not necessary to close zipOutputStream inside writer's close method.